src/sql/azdata.proposed.d.ts

```diff
@@ -2066,6 +2066,339 @@ declare module 'azdata' {
 		export function openServerFileBrowserDialog(connectionUri: string, targetPath: string, fileFilters: FileFilters[]): Thenable<string | undefined>;
 	}
 
+	//#region Chart component model types
+
+	export type ChartType = 'bar' | 'bubble' | 'doughnut' | 'horizontalBar' | 'line' | 'pie' | 'polarArea' | 'radar' | 'scatter';
+
+	export interface ModelBuilder {
+		chart<TConfig extends ChartConfiguration>(): ComponentBuilder<ChartComponent<TConfig>, ChartComponentProperties<TConfig>>;
+	}
+
+	export interface ChartComponent<TConfig extends ChartConfiguration> extends Component, ChartComponentProperties<TConfig> {
+		onDidClick: vscode.Event<ChartClickEvent>;
+	}
+
+	export type ChartClickEvent = { label: string };
```

Comment:
Am I missing something or do I not see this being implemented?

Reply:
extHostModelView uses it, but the entire data selection eventing was just stubbed out.  It's something that we'd we'll want for Query STore, but isn't finished in this PR.  I'll remove it here.